src/features/organizations/components/OrganizationSwitcher/index.tsx

```diff
@@ -0,0 +1,136 @@
+    showEmptyState,
+    showLoadingState,
+    showOrgTree,
+    showRecentOrgs,
+  } = useOrgSwitcher(orgId, searchString);
+
+  return (
+    <Box
+      sx={{
+        backgroundColor: theme.palette.background.paper,
+        borderBottomColor: showOrgSwitcher
+          ? 'transparent'
+          : theme.palette.grey[300],
+        borderBottomStyle: 'solid',
+        borderBottomWidth: 1,
+        height: showOrgSwitcher ? 'calc(100% - 130px)' : 0,
+        overflowY: 'auto',
+        position: 'absolute',
+        transition: theme.transitions.create(['borderBottomColor', 'height'], {
+          duration: theme.transitions.duration.short,
+          easing: theme.transitions.easing.sharp,
+        }),
+        width: '100%',
+        zIndex: 1000,
+      }}
+    >
+      {showLoadingState && (
+        <Box sx={{ display: 'flex', justifyContent: 'center', margin: 3 }}>
+          <CircularProgress />
+        </Box>
+      )}
+      {showEmptyState && (
+        <Box alignItems="center" display="flex" flexDirection="column">
+          <FilterListOutlined color="secondary" sx={{ fontSize: '12em' }} />
+          <Typography color="secondary">
+            {messages.sidebar.filter.noResults()}
+          </Typography>
+        </Box>
+      )}
+      {showRecentOrgs && (
+        <Box>
+          <Box
+            alignItems="center"
+            display="flex"
+            justifyContent="space-between"
+          >
+            <Typography fontSize={12} margin={1} variant="body2">
+              {messages.sidebar.recent.title().toLocaleUpperCase()}
+              {hasMatchesInRecentOrgs &&
+                ` (${messages.sidebar.filtered().toLocaleUpperCase()})`}
+            </Typography>
+            <Button
+              onClick={() => setRecentOrganizationIds([])}
+              size="small"
+              sx={{ marginRight: 2 }}
+              variant="text"
+            >
+              {messages.sidebar.recent.clear()}
+            </Button>
+          </Box>
+          <RecentOrganizations
+            onSwitchOrg={onSwitchOrg}
+            orgId={orgId}
+            recentOrganizations={
+              hasMatchesInRecentOrgs ? filteredRecentOrgs : recentOrgs
+            }
+          />
+        </Box>
+      )}
+      {showOrgTree && (
+        <Box>
+          <Typography fontSize={12} m={1} variant="body2">
+            {messages.sidebar.allOrganizations().toLocaleUpperCase()}
+            {searchString.length > 0 &&
+              ` (${messages.sidebar.filtered().toLocaleUpperCase()})`}
+          </Typography>
+          {searchString.length == 0 && (
+            <OrganizationTree
+              onSwitchOrg={onSwitchOrg}
+              orgId={orgId}
+              treeItemData={orgData}
+            />
+          )}
+          {searchString.length > 0 && (
+            <SearchResults
+              flatOrgData={flatOrgData}
+              onSwitchOrg={onSwitchOrg}
+              searchString={searchString}
```

Comment:
I don't expect this to be passed into search results, now that you have a hook here that already does the searching. Just pass in the results (not the full `flatOrgData`).

Reply:
Made this change and it causes a bug, because flatOrgData was used inside SearchResults to map through in travelling up the ancestral tree, looking for parents. Should this be moved into the hook? Or keep passing flatOrgData like this to use in looking for parents, and also pass the matches?

Reply:
Ah, I see. I guess pass the matches then. I think that's better then going back to having search happening in two places.